packages/flutter_markdown/lib/src/widget.dart

```diff
@@ -0,0 +1,211 @@
+  List<Widget> _children;
+  final List<GestureRecognizer> _recognizers = <GestureRecognizer>[];
+
+  @override
+  void didChangeDependencies() {
+    _parseMarkdown();
+    super.didChangeDependencies();
+  }
+
+  @override
+  void didUpdateConfig(MarkdownWidget oldConfig) {
+    super.didUpdateConfig(oldConfig);
+    if (config.data != oldConfig.data
+        || config.styleSheet != oldConfig.styleSheet)
+      _parseMarkdown();
+  }
+
+  @override
+  void dispose() {
+    _disposeRecognizers();
+    super.dispose();
+  }
+
+  void _parseMarkdown() {
+    final MarkdownStyleSheet styleSheet = config.styleSheet ?? new MarkdownStyleSheet.fromTheme(Theme.of(context));
+
+    _disposeRecognizers();
+
+    // TODO: This can be optimized by doing the split and removing \r at the same time
+    final List<String> lines = config.data.replaceAll('\r\n', '\n').split('\n');
+    final md.Document document = new md.Document();
+    final MarkdownBuilder builder = new MarkdownBuilder(delegate: this, styleSheet: styleSheet);
+    _children = builder.build(document.parseLines(lines));
+  }
+
+  void _disposeRecognizers() {
+    if (_recognizers.isEmpty)
+      return;
+    final List<GestureRecognizer> localRecognizers = new List<GestureRecognizer>.from(_recognizers);
+    _recognizers.clear();
+    for (GestureRecognizer recognizer in localRecognizers)
+      recognizer.dispose();
+  }
+
+  @override
+  GestureRecognizer createLink(String href) {
+    final TapGestureRecognizer recognizer = new TapGestureRecognizer()
+      ..onTap = () {
+      if (config.onTapLink != null)
+        config.onTapLink(href);
+    };
+    _recognizers.add(recognizer);
+    return recognizer;
+  }
+
+  @override
+  TextSpan formatText(MarkdownStyleSheet styleSheet, String code) {
+    if (config.syntaxHighlighter != null)
+      return config.syntaxHighlighter.format(code);
+    return new TextSpan(style: styleSheet.code, text: code);
+  }
+
+  @override
+  Widget build(BuildContext context) => config.build(context, _children);
+}
+
+/// A non-scrolling widget that parses and displays Markdown.
+///
+/// Supports all standard Markdown from the original
+/// [Markdown specification](https://daringfireball.net/projects/markdown/).
+///
+/// See also:
+///
+///  * [Markdown], which is a scrolling container of Markdown.
+///  * <https://daringfireball.net/projects/markdown/>
+class MarkdownBody extends MarkdownWidget {
+  /// Creates a non-scrolling widget that parses and displays Markdown.
+  MarkdownBody({
+    Key key,
+    String data,
+    MarkdownStyleSheet styleSheet,
+    SyntaxHighlighter syntaxHighlighter,
+    MarkdownTapLinkCallback onTapLink,
+  }) : super(
+    key: key,
+    data: data,
+    styleSheet: styleSheet,
+    syntaxHighlighter: syntaxHighlighter,
+    onTapLink: onTapLink,
+  );
+
+  @override
+  Widget build(BuildContext context, List<Widget> children) {
+    if (children.length == 1)
+      return children.single;
+    return new Column(
```

Comment:
I had BlockBody, but that failed a test.  The old code did it this way and I suspect that's what folks expect.